blacktie/tx/src/main/cpp/HttpTxManager.cxx

```diff
@@ -379,7 +376,7 @@ bool HttpTxManager::handle_request(
 					LOG4CXX_DEBUG(httptxlogger, "Return 200 OK with body: " <<
 						status << " XA status: " << res);
 					code = 200;
-					_branches.erase(branch->get_name());
+					_branches.erase(wid);
```

Comment:
When I wrote the recovery clean up code I see I added the following comment:
    
    // TODO have we caught all race conditions - recover can cause ra to call back
    // here via XAResourceManager::set_complete

I am just guessing but is it possible that we are seeing such a race condition?